plugins/ida/ida_main.ml

```diff
@@ -4,15 +4,21 @@ open Bap_future.Std
 open Bap.Std
 open Bap_ida.Std
 
-open Format
 open Result.Monad_infix
 
+open Ida_info
+open Ida_commands
+
 include Self()
 
-module Symbols = Data.Make(struct
-    type t = (string * int64 * int64) list
-    let version = "0.1"
-  end)
+module Ida_futures = struct
```

Comment:
After discussion, it's clear that I'm probably abusing the Future library here. But here's why it is the way it is:

These futures hold IDA information, which I populate in the loader if the cache does not already contain the information (caching may or may not be enabled). Thus, any read of img/symbols/brancher will first check if it's in the cache. If it's not in the cache, then it will try to read from these futures.

The correct thing here may be to wrap symbols / brancher / img in streams and then wrap any reads of these streams where I am currently doing `read_ida_future`.